clustered/common/src/main/java/org/ehcache/clustered/common/messages/ServerStoreOpMessage.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.common.messages;
+
+import org.ehcache.clustered.common.store.Chain;
+
+import java.nio.ByteBuffer;
+
+/**
+ */
+public abstract class ServerStoreOpMessage extends EhcacheEntityMessage {
+  public enum ServerStoreOp {
+    GET,
+    GETANDAPPEND,
+    APPEND,
+    REPLACE
+  }
+
+  private final String cacheId;
+  private final long key;
+
+  protected ServerStoreOpMessage(String cacheId, long key) {
+    this.cacheId = cacheId;
+    this.key = key;
+  }
+
+  @Override
+  public Type getType() {
+    return Type.SERVER_STORE_OP;
+  }
+
+  public String getCacheId() {
+    return cacheId;
+  }
+
+  public long getKey() {
+    return key;
+  }
+
+  public abstract ServerStoreOp operation();
+
+  public static class GetMessage extends ServerStoreOpMessage {
+
+    GetMessage(String cacheId, long key) {
+      super(cacheId, key);
+    }
+
+    @Override
+    public ServerStoreOp operation() {
+      return ServerStoreOp.GET;
+    }
+  }
+
+  public static class GetAndAppendMessage extends ServerStoreOpMessage {
+
+    private final ByteBuffer payload;
```

Comment:
I know this, but it need not be serializable at all.
As I said the codec would extract the appropriate bytes and send it to the server.
Neither this nor `Chain` needs to be serializable. The changes in codec will take care of it as explained n #940